app/controllers/application_controller.rb

```diff
@@ -116,6 +117,14 @@ def fix_cart_date
     cart.fix_due_date
   end
 
+  # make sure that we don't have any non-existant models in our cart that
+  # would cause errors when rendering
+  def fix_cart_items
+    session[:cart].items.each do |em, _count|
+      session[:cart].items.delete(em) unless EquipmentModel.find_by_id(em)
```

Comment:
honestly i just learned how to use collect and select while doing the reports and they are the most amazing things